pkg/ccl/importccl/exportcsv_test.go

```diff
@@ -319,108 +337,168 @@ func validateParquetFile(t *testing.T, file string, test parquetTest) error {
-		for i, col := range cols {
-			require.Equal(t, col.SchemaElement.Name, test.colNames[i])
-		}
-	}
 	if test.colFieldRepType != nil {
 		for i, col := range cols {
 			require.Equal(t, *col.SchemaElement.RepetitionType, test.colFieldRepType[i])
 		}
 	}
-	if test.vals != nil {
-		require.Equal(t, len(cols), len(test.vals[0]))
-		require.Equal(t, int(fr.NumRows()), len(test.vals))
-		count := 0
-		for {
-			row, err := fr.NextRow()
-			if err == io.EOF {
-				break
+	// Get the datums returned by the SELECT statement called in the EXPORT
+	// PARQUET statement to validate the data in the parquet file.
+	validationStmt := strings.SplitN(test.stmt, "FROM ", 2)[1]
+	test.datums, test.cols, err = ie.QueryBufferedExWithCols(
+		ctx,
+		"",
+		nil,
+		sessiondata.InternalExecutorOverride{
+			User:     security.RootUserName(),
+			Database: test.dbName},
+		validationStmt)
+	require.NoError(t, err)
+
+	for j, col := range cols {
+		require.Equal(t, col.SchemaElement.Name, test.cols[j].Name)
+	}
+	i := 0
+	for {
+		row, err := fr.NextRow()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return fmt.Errorf("reading record failed: %w", err)
+		}
+		for j := 0; j < len(cols); j++ {
+			if test.datums[i][j].ResolvedType() == types.Unknown {
+				// If we expect a null value, the row created by the parquet reader
+				// will not have the associated column.
+				_, ok := row[cols[j].SchemaElement.Name]
+				require.Equal(t, ok, false)
+				continue
 			}
+			parquetCol, err := importccl.NewParquetColumn(test.cols[j].Typ, "", false)
 			if err != nil {
-				return fmt.Errorf("reading record failed: %w", err)
+				return err
+			}
+			datum, err := parquetCol.DecodeFn(row[cols[j].SchemaElement.Name])
+			if err != nil {
+				return err
 			}
 
-			t.Logf("\n Record %v:", count)
-			for i := 0; i < len(cols); i++ {
-				if test.vals[count][i] == nil {
-					// If we expect a null value, the row created by the parquet reader
-					// will not have the associated column.
-					_, ok := row[cols[i].SchemaElement.Name]
-					require.Equal(t, ok, false)
+			tester := test.datums[i][j]
+			switch tester.ResolvedType().Family() {
+			case types.DateFamily:
+				// pgDate.orig property doesn't matter and can cause the test to fail
+				require.Equal(t, tester.(*tree.DDate).Date.UnixEpochDays(),
+					datum.(*tree.DDate).Date.UnixEpochDays())
+			case types.JsonFamily:
+				// Only the value of the json object matters, not that additional properties
+				require.Equal(t, tester.(*tree.DJSON).JSON.String(),
+					datum.(*tree.DJSON).JSON.String())
+			case types.FloatFamily:
+				if tester.(*tree.DFloat).String() == "NaN" {
+					// NaN != NaN, therefore stringify the comparison.
+					require.Equal(t, "NaN", datum.(*tree.DFloat).String())
 					continue
 				}
-				v := row[cols[i].SchemaElement.Name]
-				decodedV := decodeEl(t, v)
-				t.Logf("\t %v", decodedV)
-				require.Equal(t, test.vals[count][i], decodedV)
+				fallthrough
```

Comment:
[nit] I would slightly prefer just duplicating the require.Equal than using fallthrough so it is trivial to add new cases.